functions/post_form_url_encoded.go

```diff
@@ -80,79 +74,17 @@ func argsToRequestBody(existingHeaders map[string]string, args []string) url.Val
 	return values
 }
 
-// Retrieves the JSON element (specified by a dot-delimited string)
-func retrieveElementTree(jsonContents []byte, str string) ([]byte, error) {
-	elements := strings.Split(str, ".")
-	var err error
-	for _, element := range elements {
-		jsonContents, err = retrieveElement(jsonContents, element)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	return jsonContents, nil
-}
-
-// Retrieve the element by parsing into a map for a JSON object or slice for a JSON array.
-func retrieveElement(jsonContents []byte, element string) ([]byte, error) {
-	// Parse into a map for a JSON object
-	jsonObject := make(map[string]json.RawMessage)
-	err := json.Unmarshal(jsonContents, &jsonObject)
-	if err != nil {
-		// It must be a JSON array, so parse into a slice
-		jsonArray := make([]json.RawMessage, 0)
-		err = json.Unmarshal(jsonContents, &jsonArray)
-		if err != nil {
-			return nil, err
-		}
-
-		i, err := strconv.Atoi(element)
-		if err != nil {
-			return nil, fmt.Errorf("invalid index for JSON array: %v", element)
-		}
-
-		if len(jsonArray) <= i || i < 0 {
-			return nil, fmt.Errorf("JSON array index out of bounds: %v", element)
-		}
-
-		return jsonToBytes(jsonArray[i])
+func retrieveElement(data []byte, field string) (string, error) {
+	if !gjson.ValidBytes(data) {
+		return "", fmt.Errorf("invalid JSON")
 	}
 
-	if value, ok := jsonObject[element]; ok {
-		return jsonToBytes(value)
+	var result string
+	if field == "" {
```

Comment:
I think this needs to encapsulate the JSON validation above for situations where the response is not JSON